docs/proposed/psa-driver-interface.md

```diff
@@ -345,6 +347,90 @@ Transparent drivers may provide the following key management entry points:
 
 Transparent drivers are not involved when importing, exporting, copying or destroying keys, or when generating or deriving symmetric keys.
 
+### Random generation entry points
+
+A transparent driver may provide an operation family that allows it to generate random data. The random generation mechanism must obey the following requirements:
+
+* The random output must be of cryptographic quality, with a uniform distribution. Therefore, if the random generator includes an entropy source, this entropy source must be fed through a CSPRNG (cryptographically secure pseudo-random number generator).
+* Random generation is expected to be fast. (If a device can provide entropy but is slow at generating random data, declare it as an [entropy driver](#entropy-collection-entry-point) instead.)
+* The random generator must be able to incorporate entropy provided by an outside source.
```

Comment:
What of the cases where a hardware RNG doesn't accept external input?

Reply:
In general, that driver would only be suitable as one entropy source. It's a matter of security design: if there are multiple entropy sources, the RNG should remain strong even if all but one sources are compromised.

I do think there is a fairly common design space where there is a single peripheral that includes an entropy source and a DRBG, and the integrator does not wish to save entropy to storage. In this case, `add_entropy` is not needed.

Looking at what I've written, I think that `add_entropy` can be made optional, but a driver without an `add_entropy` entry point will be less portable: it can only be integrated on platforms without other entropy sources and without saved entropy.

Reply:
Can you clarify that in the spec, then? It currently doesn't cater to those devices that have a single entropy source and CSPRNG prepackaged in a hardware peripheral, with no way of accessing these separately.

Reply:
Done.